File: app/src/main/java/org/androidforfun/framework/Sound.java
/*
 *  Copyright (C) 2016 Mario Zechner
 *  This file is part of Framework for book Beginning Android Games.
 *
 *  This library is free software: you can redistribute it and/or modify
 *  it under the terms of the GNU General Public License as published by
 *  the Free Software Foundation, either version 3 of the License, or
 *  (at your option) any later version.
 *
 *  This library is distributed in the hope that it will be useful,
 *  but WITHOUT ANY WARRANTY; without even the implied warranty of
 *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 *  GNU General Public License for more details.
 *
 *  You should have received a copy of the GNU General Public License.
 */
package org.androidforfun.framework;

/** <p>
 * A Sound is a short audio clip that can be played numerous times in parallel. It's completely
 * loaded into memory so only load small audio files. Call the {@link #dispose()} method when
 * you're done using the Sound.
 * </p>
 *
 * <p>
 * Sound instances are created via a call to {@link Audio#newSound(String)}.
 * </p>
 *
 * @author mzechner
 */
public interface Sound {
    /**
     * Plays the sound. If the sound is already playing, it will be played again, concurrently.
     * @param volume the volume in the range [0,1]
     * @return the id of the sound instance if successful, or -1 on failure.
     */
    void play(float volume);

    /**
     * Releases all the resources.
     */
    void dispose();
}

File: app/src/main/kotlin/org/code4projects/framework/Sound.kt
/*
 * Copyright (c) 2016-2026 Salvatore D'Angelo
 * Licensed under the MIT License. See the LICENSE file in the project root.
 */
package org.code4projects.framework

/**
 * A Sound is a short audio clip that can be played numerous times in parallel. It's completely
 * loaded into memory so only load small audio files. Call the [dispose] method when
 * you're done using the Sound.
 *
 * Sound instances are created via a call to [Audio.newSound].
 *
 * @author mzechner
 */
interface Sound {
    /**
     * Plays the sound. If the sound is already playing, it will be played again, concurrently.
     * @param volume the volume in the range [0,1]
     */
    fun play(volume: Float)

    /**
     * Releases all the resources.
     */
    fun dispose()
}
